paddle/fluid/framework/new_exec.h

```diff
@@ -112,6 +156,36 @@ struct OpFuncNode {
   OpKernelComputeFunc kernel_func_;
 };
 
+struct CostInfo {
+  CostInfo() { Reset(); }
+
+  void Reset() {
+    total_time = 0.;
+    host_memory_bytes = 0;
+    device_memory_bytes = 0;
+  }
+
+  void Start(bool is_exclude = false) {
+    auto& timer = is_exclude ? exclude_timer : cost_timer;
+    timer.Start();
+  }
+  void Pause(bool is_exclude = false) {
+    auto& timer = is_exclude ? exclude_timer : cost_timer;
+    timer.Pause();
+    if (is_exclude) {
+      total_time -= cost_timer.ElapsedMS();
+    } else {
+      total_time += cost_timer.ElapsedMS();
+    }
+  }
+
+  double total_time;           // ms
+  size_t host_memory_bytes;    // bytes
+  size_t device_memory_bytes;  // bytes
+  platform::Timer cost_timer;
+  platform::Timer exclude_timer;
```

Comment:
What's the function of the exclude_timer? Besides, Why does CostInfo contain Timer? It will be better to decouple these two or rename CostInfo to something else such as CostCollector.

Reply:
Thanks, fixed